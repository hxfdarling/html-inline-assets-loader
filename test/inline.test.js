
import 'jest-extended';

const { compile, evaluated } = require('./helpers');

describe('inline loader', () => {
  it('should inline html/js/css', async () => {
    const stats = await compile('inline.html');

    expect(stats.compilation.warnings).toBeArrayOfSize(0);
    expect(stats.compilation.errors).toBeArrayOfSize(0);

    const { modules } = stats.toJson();
    const [module] = modules;
    expect(module.source).toMatchSnapshot('module');
    expect(evaluated(module.source, modules))
      .toMatchSnapshot('module (evaluated)');
  });
});
