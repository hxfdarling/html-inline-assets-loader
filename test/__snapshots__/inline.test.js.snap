// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline loader should inline html/js/css: module (evaluated) 1`] = `
"<!DOCTYPE html><html><head>
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
<meta charset=\\"utf-8\\">
<meta name=\\"format-detection\\" content=\\"telephone=no\\">
<meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1.0,maximum-scale=1.0,user-scalable=no\\">

    <title>imt application</title>
    <style type=\\"text/css\\">body{
  background:red;
}</style>
    <script>\\"use strict\\";

console.log('inline');</script>
  </head>
  <body>

</body></html>"
`;

exports[`inline loader should inline html/js/css: module 1`] = `"module.exports = \\"<!DOCTYPE html><html><head>\\\\n    <meta http-equiv=\\\\\\"X-UA-Compatible\\\\\\" content=\\\\\\"IE=edge\\\\\\">\\\\n<meta charset=\\\\\\"utf-8\\\\\\">\\\\n<meta name=\\\\\\"format-detection\\\\\\" content=\\\\\\"telephone=no\\\\\\">\\\\n<meta name=\\\\\\"viewport\\\\\\" content=\\\\\\"width=device-width,initial-scale=1.0,maximum-scale=1.0,user-scalable=no\\\\\\">\\\\n\\\\n    <title>imt application</title>\\\\n    <style type=\\\\\\"text/css\\\\\\">body{\\\\n  background:red;\\\\n}</style>\\\\n    <script>\\\\\\"use strict\\\\\\";\\\\n\\\\nconsole.log('inline');</script>\\\\n  </head>\\\\n  <body>\\\\n\\\\n</body></html>\\";"`;
